engine/src/main/java/org/hibernate/search/engine/spi/DocumentBuilderIndexedEntity.java

```diff
@@ -661,10 +661,33 @@ else if ( currentFieldValue instanceof Map ) {
 					if ( fieldMetadata.hasFacets() ) {
 						faceting.enableFacetProcessing();
 						for ( FacetMetadata facetMetadata : fieldMetadata.getFacetMetadata() ) {
-							if ( multiValuedParent ) {
+							boolean multivalued = multiValuedParent;
+
+							if ( member.isCollection() && currentFieldValue instanceof Collection ) {
+								multivalued = true;
+								for ( Object element : (Collection<?>) currentFieldValue ) {
+									addFacetDocValues( document, fieldMetadata, facetMetadata, element );
+								}
+							}
+							else if ( member.isCollection() && currentFieldValue instanceof Map ) {
+								multivalued = true;
+								for ( Object element : ((Map<?,?>) currentFieldValue).values() ) {
+									addFacetDocValues( document, fieldMetadata, facetMetadata, element );
+								}
+							}
+							else if ( member.isArray() ) {
+								multivalued = true;
+								for ( Object element : (Object[]) currentFieldValue ) {
```

Comment:
See my other PR: https://github.com/hibernate/hibernate-search/pull/1274

`@Facet` does not work if we don't also have a `@Field`. Which means supporting arrays of primitives for facets makes little sense if we have no support for arrays of primitives in the field bridges.

Plus, I wanted to discuss the whole solution of `Array.*` in a single PR, because it's likely to raise some objections.